std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoSecretHelper.java

```diff
@@ -177,14 +179,42 @@ public EnsoHttpResponse reconstructResponseFromCachedStream(
   }
 
   private static EnsoHTTPResponseCache getOrCreateCache() {
-    if (cache == null) {
-      cache = new EnsoHTTPResponseCache();
+    if (getCache() instanceof EnsoHTTPResponseCache httpCache) {
+      return httpCache;
+    } else {
+      var module =
+          Context.getCurrent()
+              .eval(
+                  "enso",
+                  """
+      import Standard.Base.Runtime.Ref.Ref
+      import Standard.Base.Data.Boolean.Boolean
+
+      type Cache
+          private Value ref:Ref
+
+          new obj -> Cache =
+            ref = Ref.new obj Boolean.False
```

Comment:
Changing `Boolean.False` to `Boolean.True` would _allow GC_ of the cache, @GregoryTravis,